core/vm/instructions.go

```diff
@@ -472,7 +467,11 @@ func opExtCodeCopy(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext)
 	}
 	addr := common.Address(a.Bytes20())
 	if interpreter.evm.Accesses != nil {
-		log.Warn("setting witness values for extcodecopy is not currently implemented")
+		code := interpreter.evm.StateDB.GetCode(addr)
+		paddedCodeCopy, copyOffset, nonPaddedCopyLength := getDataAndAdjustedBounds(code, uint64CodeOffset, length.Uint64())
+		cb := codeBitmap(code)
+		touchEachChunksAndChargeGas(copyOffset, nonPaddedCopyLength, addr[:], code, &cb, interpreter.evm.Accesses)
```

Comment:
I don't think you actually need to do the analysis here, nor to charge the gas at this point, or to know what in the code is pushdata and what is code.

As long as you know the start offset, you can divide it into 31-byte chunks. I might be missing something, so we can have a call about it.

Reply:
Ah ok, now I get it: one needs to know what the value of the pushdata offset it. Hmmm....